featuretools/tests/primitive_tests/natural_language_primitives_tests/test_num_words.py

```diff
@@ -0,0 +1,76 @@
+import numpy as np
+import pandas as pd
+
+from featuretools.primitives import NumWords
+from featuretools.tests.primitive_tests.utils import (
+    PrimitiveT,
+    find_applicable_primitives,
+    valid_dfs,
+)
+
+
+class TestNumberOfUniqueWords(PrimitiveT):
+    primitive = NumWords
+
+    def test_general(self):
+        x = pd.Series(
+            [
+                "test test test test",
+                "test TEST test TEST,test test test",
+                "and subsequent lines...",
+            ],
+        )
+        expected = pd.Series([4, 6, 3])
+        actual = self.primitive().get_function()(x)
+        pd.testing.assert_series_equal(actual, expected, check_names=False)
+
+    def test_special_characters_and_whitespace(self):
+        x = pd.Series(["50% 50 50% \t\t\t\n\n", "a test* test"])
```

Comment:
What about numbers like `$5,031.40` is that one word? I think maybe we should test for that type of scenario as well.